src/components/feedback/Dialog.tsx

```diff
@@ -86,13 +115,20 @@ const DialogNext = function Dialog({
   });
 
   const dialogDescriptionId = useUniqueId('dialog-description');
+  const Wrapper = useMemo(
+    () => TransitionComponent ?? Fragment,
+    [TransitionComponent]
+  );
```

Comment:
I'm not sure if it's worth memoizing this. Maybe `const Wrapper = TransitionComponent ?? Fragment` would suffice 🤷🏼‍♂️